components/bl-rating-component/src/index.js

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useRef, useState } from 'react';
+
+import { useRatingClassList } from './hooks/useRatingClassList';
+import { iconParams } from './utils/icon-params';
+import { iconsList } from './utils/icons-list.js';
+import { roundValueToPrecision } from './utils/number';
+import { RatingIcon } from './rating-icon';
+
+const RATING_PRECISION = 0.5;
+
+export default function RatingComponent({
+  component, eventHandlers,
+}) {
+  const {
+    display,
+    disabled,
+    icon,
+    iconColor,
+    defaultValue,
+    iconsAmount,
+    classList,
+  } = component;
+  const { onRatingChange } = eventHandlers;
+  const classes = useRatingClassList(disabled, classList);
+
+  const rootRef = useRef();
+  const [ratingValue, setRatingValue] = useState(defaultValue);
+  const [hoverValue, setHoverValue] = useState();
+  const icons = iconsList(iconsAmount);
+
+  const value = hoverValue || ratingValue;
+
+  if (!display) {
```

Comment:
return null always must be after all hooks calling

https://bobbyhadz.com/blog/react-rendered-more-hooks-than-during-previous-render